mautic/mautic
# Set the notification button nullable to fix API library unit tests

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | true
| New feature? | false
| Related user documentation PR URL | null
| Related developer documentation PR URL | null
| Issues addressed (#s or URLs) | null
| BC breaks? | false
| Deprecations? | false

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
The PR https://github.com/mautic/mautic/pull/2980 caused the PHP API library unit tests to fail with:
```
1) Mautic\Tests\Api\NotificationsTest::testCreateGetAndDelete
An exception occurred while executing 'INSERT INTO push_notifications (is_published, date_added, created_by, created_by_user, date_modified, modified_by, modified_by_user, checked_out, checked_out_by, checked_out_by_user, name, description, lang, url, heading, message, button, notification_type, publish_up, publish_down, read_count, sent_count, category_id) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)' with params [1, "2017-01-13 13:34:57", 1, "John Linhart", null, null, null, null, null, null, "test", null, "en", null, "API test heading", "API test message", null, "template", null, null, 0, 0, null]:

SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'button' cannot be null
Failed asserting that true is false.

tests/Api/MauticApiTestCase.php:41
tests/Api/MauticApiTestCase.php:51
tests/Api/NotificationsTest.php:37
```
This PR makes the notification button nullable.

#### Steps to test this PR:
1. Apply this PR.
2. Run the migrations.
3. Repeat the tests. All should be successful.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Run `phpunit` on https://github.com/mautic/api-library
